src/app/view/components/dot-crumbtrail/service/dot-crumbtrail.service.spec.ts

```diff
@@ -85,11 +132,194 @@ describe('DotCrumbtrailService', () => {
         });
 
         service = testbed.get(DotCrumbtrailService);
-    });
 
-    it('should take the current url', () => {
         service.crumbTrail$.subscribe((crumbs) => {
-            console.log('CRUMBS', crumbs);
+            if (!firstCrumb) {
+                firstCrumb = crumbs;
+            } else {
+                secondCrumb = crumbs;
+            }
         });
+
+        tick();
+    }));
+
+    it('should take the current url from Router', () => {
+        expect(firstCrumb).toEqual([
+            {
+                label: 'menu',
+                url: '#/menulink/first_portlet'
+            },
+            {
+                label: 'Potlet Label',
+                url: '#/menulink/portlet'
+            }
+        ]);
     });
+
+    it('Should take url from NavegationEnd event', fakeAsync(() => {
+        dotNavigationServiceMock.navigationEnd.next(
+            {
+                url: '/first_portlet',
+                urlAfterRedirects: '/first_portlet',
+                id: 1,
+            }
+        );
+
+        tick();
+
+        expect(secondCrumb).toEqual([
+            {
+                label: 'menu',
+                url: '#/menulink/first_portlet'
+            },
+            {
+                label: 'First Portlet Label',
+                url: '#/menulink/first_portlet'
+            }
+        ]);
+    }));
+
+    it('Should ignore c prefix', fakeAsync(() => {
+        dotNavigationServiceMock.navigationEnd.next(
+            {
+                url: '/first_portlet',
+                urlAfterRedirects: '/first_portlet',
+                id: 1,
+            }
+        );
+
+        tick();
+
+        expect(secondCrumb).toEqual([
+            {
+                label: 'menu',
+                url: '#/menulink/first_portlet'
+            },
+            {
+                label: 'First Portlet Label',
+                url: '#/menulink/first_portlet'
+            }
+        ]);
+    }));
+
+    it('Should exclude URL', fakeAsync(() => {
+        dotNavigationServiceMock.navigationEnd.next(
+            {
+                url: '/content-types-angular/create/content',
+                urlAfterRedirects: '/content-types-angular/create/content',
+                id: 1,
+            }
+        );
+
+        tick();
+
+        expect(secondCrumb).toEqual([
+            {
+                label: 'Types & Tag',
+                url: '#/content-types-angular'
+            },
+            {
+                label: 'Content Types',
+                url: '#/content-types-angular'
+            }
+        ]);
+    }));
+
+    it('Should take content types data', fakeAsync(() => {
+        mockActivatedRoute.root = {
```

Comment:
too much dupe code, you might want to create a utlity function to generate this for all cases.